api/src/main/java/org/apache/iceberg/ScanTask.java

```diff
@@ -31,6 +31,15 @@ default long sizeBytes() {
     return 4 * 1028 * 1028; // 4 MB
   }
 
+  /**
+   * The estimated number of rows produced by this scan task.
+   *
+   * @return the estimated number of produced rows
+   */
+  default long rowsCount() {
```

Comment:
nit: should it be `estimatedRowCount`?

Reply:
I went back and forth it initially. I'll change if you think that's cleaner.

Reply:
Done.